jaypeasee/sorting-hat-quiz
# Feature - Quiz Logic

#### What’s this PR do?
This PR builds out all the quiz logic. Specifically, it:
1. fetches `housesData` on `App` load.
2. creates state on the input of the `Name` component.
3. creates state on the answers of the `Quiz` component and passes data back to App to update state.
4. determines the user house based on quiz answers.
5. fetches `characterData` on `Result` and parses it based on the user's determined house.
6. Resets the quiz if the user decides to take it again.

#### Where should the reviewer start?
* `App.js`
* `Name.js`
* `Question.js`
* `Result.js`

#### How should this be manually tested?
1. Run `npm start`
2. Open React dev tools 
3. Take the quiz and look at the components above to see how state changes in each of them.

#### Any background context you want to provide?
This completes the majority of the logic for the quiz. Next steps will be to interpolate the data into the results page and style it a bit more.

#### What are the relevant tickets?
* #13 
* #14
* #15
* #16
* #17
